mist-lib/src/services/stp-utils.tests.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable no-magic-numbers, no-undefined */
+'use strict'
+
+const path = require('path')
+
+const tableFilePath = path.resolve(__dirname, 'test-data/mist3-sig-trans-2017.test.csv')
+const stpUtils = require('./stp-utils.js')
+
+describe('Signal Transduction Service', function() {
+	it('parseCSVTable', function() {
+		let expectedPfql = [
+			{
+				rules: [
+					{
+						Npos: [
+							{
+								resource: 'pfam30',
+								feature: 'PAS'
+							}
+						]
+					}
+				]
+			},
+			{
+				rules: [
+					{
+						Npos: [
+							{
+								resource: 'pfam30',
+								feature: 'PAS_2'
+							}
+						]
+					}
+				]
+			},
+			{
+				rules: [
+					{
+						Npos: [
+							{
+								resource: 'pfam30',
+								feature: 'BTAD'
+							}
+						]
+					}
+				]
+			},
+			{
+				rules: [
+					{
+						Npos: [
+							{
+								resource: 'pfam30',
+								feature: 'HisKA'
+							}
+						]
+					}
+				]
+			},
+			{
+				rules: [
+					{
+						Npos: [
+							{
+								resource: 'pfam30',
+								feature: 'HTH_10'
+							}
+						]
+					}
+				]
+			}
+		]
+		return stpUtils.parseCSVTable(tableFilePath).then(function(results) {
+			expect(results.length).eql(expectedPfql.length)
+			for (let i = 0; i < results.length; i++)
+				expect(results[i].rules).eql(expectedPfql[i].rules)
```

Comment:
`eql` is for performing deep object comparison. When comparing primitives like numbers, please prefer `equal` instead. That being said, given that we want `results` to exactly match `expectedPfql`, we can do the above three lines with just this:

`expect(results).eql(expectedPfql)`

Reply:
Each element of an array of rules (expectPfql) must match a specific field in the object of same index in the results array. I don't think chai can do this in batch. I am keeping as is unless someone help me out here.